plutus-core/untyped-plutus-core/test/Generators.hs

```diff
@@ -80,6 +84,25 @@ propParser = testPropertyNamed "Parser" "parser" $ property $ do
             :: T.Text -> Either ParserErrorBundle (Program Name DefaultUni DefaultFun SrcSpan)
         parseProg = runQuoteT . parseProgram
 
+-- | The `SrcSpan` of a parsed `Term` should not including trailing whitespaces.
+propTermSrcSpan :: TestTree
+propTermSrcSpan = testPropertyNamed
+    "parser captures ending positions correctly"
+    "propTermSrcSpan"
+    . property
+    $ do
```

Comment:
You could write this test generically as a test that takes a program generator, a pretty-printer, and a parser. Not sure if it's worth it.

Reply:
Good point.

Reply:
It'd also need to take a `termAnn` function as the parameter since it is different for PIR/TPLC/UPLC. So it's probably not worth it - the amount of reused code is limited.